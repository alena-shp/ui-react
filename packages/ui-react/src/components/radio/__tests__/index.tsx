import React from 'react';
import { create } from 'react-test-renderer';

import { Radio } from '..';

test('Default', () => {
    const tree = create(
        <Radio
            tabIndex={1}
            name="default"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Disabled', () => {
    const tree = create(
        <Radio
            disabled={true}
            tabIndex={1}
            name="default"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Default Checked', () => {
    const tree = create(
        <Radio
            checked={true}
            tabIndex={1}
            name="default"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Default Checked Disabled', () => {
    const tree = create(
        <Radio
            checked={true}
            disabled={true}
            tabIndex={1}
            name="default"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('With Description', () => {
    const tree = create(
        <Radio
            description="Desc"
            tabIndex={1}
            name="default"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('With Id', () => {
    const tree = create(
        <Radio
            id="id"
            name="default"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});
