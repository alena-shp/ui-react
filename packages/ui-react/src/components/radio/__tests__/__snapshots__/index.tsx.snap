// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default 1`] = `
<label
  className="radio"
>
  <input
    aria-checked={false}
    className="radio__input"
    name="default"
    role="radio"
    tabIndex={1}
    type="radio"
  />
  <p
    className="typography typography_size-body typography_color-basic typography_no-spacing"
  />
</label>
`;

exports[`Default Checked 1`] = `
<label
  className="radio"
>
  <input
    aria-checked={true}
    checked={true}
    className="radio__input"
    name="default"
    role="radio"
    tabIndex={1}
    type="radio"
  />
  <p
    className="typography typography_size-body typography_color-basic typography_no-spacing"
  />
</label>
`;

exports[`Default Checked Disabled 1`] = `
<label
  className="radio radio_disabled"
>
  <input
    aria-checked={true}
    checked={true}
    className="radio__input"
    disabled={true}
    name="default"
    role="radio"
    tabIndex={-1}
    type="radio"
  />
  <p
    className="typography typography_size-body typography_color-basic typography_no-spacing"
  />
</label>
`;

exports[`Disabled 1`] = `
<label
  className="radio radio_disabled"
>
  <input
    aria-checked={false}
    className="radio__input"
    disabled={true}
    name="default"
    role="radio"
    tabIndex={-1}
    type="radio"
  />
  <p
    className="typography typography_size-body typography_color-basic typography_no-spacing"
  />
</label>
`;

exports[`With Description 1`] = `
<label
  className="radio radio_description"
>
  <input
    aria-checked={false}
    className="radio__input"
    name="default"
    role="radio"
    tabIndex={1}
    type="radio"
  />
  <p
    className="typography typography_size-body typography_color-basic typography_no-spacing"
  />
  <span
    className="radio__description"
  >
    Desc
  </span>
</label>
`;

exports[`With Id 1`] = `
<label
  className="radio"
  id="id"
>
  <input
    aria-checked={false}
    className="radio__input"
    id="id-input"
    name="default"
    role="radio"
    tabIndex={0}
    type="radio"
  />
  <p
    className="typography typography_size-body typography_color-basic typography_no-spacing"
  />
</label>
`;
